src/Microsoft.Cci.Extensions/Writers/CSharp/CSDeclarationWriter.Methods.cs

```diff
@@ -213,19 +225,31 @@ private void WriteMethodBody(IMethodDefinition method)
                 else
                     Write($"System.PlatformNotSupportedException(\"{_platformNotSupportedExceptionMessage}\"); ");
             }
-            else if (method.ContainingTypeDefinition.IsValueType && method.IsConstructor)
+            else if (NeedsMethodBodyForCompilation(method))
             {
-                // Structs cannot have empty constructors so we need to output this dummy body
-                Write("throw null;");
-            }
-            else if (!TypeHelper.TypesAreEquivalent(method.Type, method.ContainingTypeDefinition.PlatformType.SystemVoid))
-            {
-                WriteKeyword("throw null;");
+                Write("throw null; ");
             }
 
             WriteSymbol("}");
         }
 
+        private bool NeedsMethodBodyForCompilation(IMethodDefinition method)
```

Comment:
Nit: You can keep it simple and just emit `throw null` all the time. `throw null` vs. empty body has no impact on the binary size - the binary size will be same in both cases.

Reply:
I guess that is true. I wasn't as concerned wit the binary size as much as I was the source code size. Ideally I would just be empty bodies all the time. Perhaps the new compiler /refonly flag might allow that I should check.

Reply:
" => throw null; "    is not a lot of code :-) 

Allowing empty bodies (or no bodies at all - like in interfaces) was considered, but I am not sure it was done (yet?). 
CC: @jcouv